prov/rxm/src/rxm_cq.c

```diff
@@ -485,9 +488,31 @@ ssize_t rxm_cq_handle_large_data(struct rxm_rx_buf *rx_buf)
 				rx_buf, rx_buf->recv_entry->total_len);
 		}
 		total_recv_len -= copy_len;
+		if (RXM_MR_RAW(rx_buf->ep->msg_info)) {
+			size_t raw_key_size = rx_buf->ep->msg_info->domain_attr->mr_key_size;
+			char* raw_key_base = rx_buf->pkt.data + sizeof(struct rxm_rndv_hdr);
+			size_t raw_key_hdr_size = sizeof(struct rxm_rndv_raw_key_hdr) + raw_key_size;
+			struct rxm_rndv_raw_key_hdr* raw_key_hdr =
+			    (struct rxm_rndv_raw_key_hdr*)(raw_key_base + i * raw_key_hdr_size);
+			ret = fi_mr_map_raw(msg_domain, raw_key_hdr->base_addr,
+			                    raw_key_hdr->raw_key, raw_key_size,
+			                    &rx_buf->rndv_hdr->iov[i].key, /* flags = */ 0);
+			if (OFI_UNLIKELY(ret)) {
+				FI_WARN(&rxm_prov, FI_LOG_CQ, "fi_mr_map_raw failed\n");
+				rxm_cq_write_error(rx_buf->ep->util_ep.rx_cq,
+				                   rx_buf->ep->util_ep.rx_cntr,
+				                   rx_buf->recv_entry->context, ret);
+				return ret;
+			}
+		}
```

Comment:
This is a significant amount of code to add to this function.  I would rather see this isolated, maybe even have separate functions for handling raw mr's, so we lessen the impact when they aren't needed.